huaweicloud/services/swr/resource_huaweicloud_swr_image_permissions.go

```diff
@@ -0,0 +1,377 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product SWR
+// ---------------------------------------------------------------
+
+package swr
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/identity/v3.0/users"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceSwrImagePermissions() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSwrImagePermissionsCreate,
+		UpdateContext: resourceSwrImagePermissionsUpdate,
+		ReadContext:   resourceSwrImagePermissionsRead,
+		DeleteContext: resourceSwrImagePermissionsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"organization": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the name of the organization.`,
+			},
+			"repository": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the name of the repository.`,
+			},
+			"users": {
+				Type:        schema.TypeList,
+				Elem:        ImagePermissionsUserSchema(),
+				Required:    true,
+				Description: `Specifies the users to access to the image (repository).`,
+			},
+			"self_permission": {
+				Type:        schema.TypeList,
+				Elem:        ImagePermissionsUserSchema(),
```

Comment:
it better to define the Elem with Computed only fields.

Reply:
done